ts/src/test/Exchange/test.fetchOHLCV.ts

```diff
@@ -11,16 +12,54 @@ async function testFetchOHLCV (exchange, skippedProperties, symbol) {
     if (!exchange.inArray (chosenTimeframeKey, timeframeKeys)) {
         chosenTimeframeKey = timeframeKeys[0];
     }
-    const limit = 10;
-    const duration = exchange.parseTimeframe (chosenTimeframeKey);
-    const since = exchange.milliseconds () - duration * limit * 1000 - 1000;
+    const limit = 50;
+    const durationMs = exchange.parseTimeframe (chosenTimeframeKey) * 1000;
+    const since = exchange.milliseconds () - durationMs * limit - 1000;
     const ohlcvs = await exchange.fetchOHLCV (symbol, chosenTimeframeKey, since, limit);
     assert (Array.isArray (ohlcvs), exchange.id + ' ' + method + ' must return an array, returned ' + exchange.json (ohlcvs));
     const now = exchange.milliseconds ();
     for (let i = 0; i < ohlcvs.length; i++) {
         testOHLCV (exchange, skippedProperties, method, ohlcvs[i], symbol, now);
     }
-    // todo: sorted timestamps check
+    const logText = testSharedMethods.logTemplate (exchange, method, ohlcvs);
+    // check boundaries
+    const barsLength = ohlcvs.length;
+    const barsLengthAboveZero = barsLength > 0;
+    // ensure bars amount is less then limit
+    assert (barsLength <= limit, 'Returned bars amount (' + barsLength.toString () + ') is more than requested (' + limit.toString () + ')' + logText);
+    // ensure bars amount is more than zero
+    if (!('compareAmountToZero' in skippedProperties)) {
+        assert (barsLengthAboveZero, 'Returned bars amount should be more then zero' + logText);
+    }
+    // if no bars were returned, we can't do any more checks
+    if (!barsLengthAboveZero) {
+        return;
+    }
+    const minActrualTs = ohlcvs[0][0];
+    const maxActualTs = ohlcvs[barsLength - 1][0];
+    const maxExpectedTs = minActrualTs + durationMs * limit;
+    // ensure that timestamps are greaterOrEqual than since
+    if (!('compareTimestampToSince' in skippedProperties)) {
+        assert (minActrualTs >= since, 'Returned bars earliest timestamp (' + minActrualTs.toString () + ') is before than requested since (' + since.toString () + ')' + logText);
+    }
+    // ensure last timestamp is under than requested since + limit * duration
+    if (limit !== undefined) {
+        assert (maxActualTs <= maxExpectedTs, 'Returned bars maximum timestamp (' + maxActualTs.toString () + ') is greater than expected limit timestamp (' + maxExpectedTs.toString () + ')' + logText);
+    }
+    // ensure current bar is greaterOrEqual than previous one by duration
+    if (!('compareDuration' in skippedProperties)) {
```

Comment:
@ttodua The `since` and `limit` checks I agree because we might be doing something wrong there, but I'm just saying that ccxt does not control what the API responds,  so if they emit some invalid candles, what can ccxt do about it? We will skip it from tests but users are going to call `fetchOHLCV` any way they have no idea how we run our tests so in practice does not seem to make a difference

Reply:
but if any exchange emits some invalid data for anything, what we do in general? there is nothing we can do, user just gets invalid data as it is happening till date.  however, we are not aiming to fix particular exceptions here or there (which might  already be happening). in "skips.json" we would collect all buggeous exchagnes, so we will know which exchanges might be misbehaving, instead of getting sporadic reports from users saying that here or there something does not work. why are we leaving those bugs to be reported back to us by users? we would have them ready in skips.json.
but ofc, if you think otherwise, please lmk which part should I remove specifically from this pr

Reply:
@ttodua Don't get me wrong, I think tests are very important, I'm just questioning their efficacy when they test things that ccxt has no control over, ccxt wraps the API of certain exchanges but it cannot change it .

Reply:
sure, the part which I couldnt understand was , what does _'ccxt has no control over'_ means in this case. I mean, that according to this PR, most of exchanges run well and pass tests. If there appears somewhat exchagne that has crappy engine, we add it in "skip-json" for that particular misbehavior, and then we will always remind whether this or that exchange return crappy/buggoues data, and it will also be reference for any users which face some bugs/glitches in their apps, thinking that "ccxt" has bugs. 
we would always know whether this or that bug comes from that exchange or not, as we would have detailed tests and skips

Reply:
so, what I can't understand well is that what harm these detailed tests would cause. we will just (very rarely) might face somewhat exchange failing, and we can add it in skip pretty easily. this PR passed well with these new detailed tests.

Reply:
> so, what I can't understand well is that what harm these detailed tests would cause. we will just (very rarely) might face somewhat exchange failing, and we can add it in skip pretty easily. this PR passed well with these new detailed tests.

 @ttodua I would not say "harm" but as with everything, it adds some complexity but I'm ok with them getting merged don't worry

Reply:
@carlosmiei lmk if something missing. the tests seem to pass for all exchanges and I think it's good. if ever smth glitch ever happens, I'll be here to look at that.